hydra-node/src/Hydra/Node.hs

```diff
@@ -114,22 +114,28 @@ stepHydraNode tracer node = do
   traceWith tracer $ BeginEvent{by = party, eventId, event = queuedEvent}
   outcome <- atomically (processNextEvent node queuedEvent)
   traceWith tracer (LogicOutcome party outcome)
-  effs <- case outcome of
+  effs <- handleOutcome e outcome
+  mapM_ (uncurry $ flip $ processEffect node tracer) $ zip effs (map (eventId,) [0 ..])
+  traceWith tracer EndEvent{by = party, eventId}
+ where
+  handleOutcome e = \case
```

Comment:
We could have a `collectEffects` which would collect and _flatten_ all the effects of an Outcome.

We could have a `wait` function which would wait for all the `Wait` of an Outcome.

Let's keep that idea of having a co-algebra in the back of our head if we happen to need it.